lib/Fencer/Rules.hs

```diff
@@ -65,14 +70,27 @@ loadRulesFromDirectory
     let filteredFiles = if ignoreDotFiles
         then filter (not . isDotFile) files
         else files
-    (errs, rules) <- partitionEithers <$> mapM loadFile filteredFiles
-    pure $ if (null @[] errs) then Right rules else Left errs
+    validatePotentialDomains <$> mapM loadFile filteredFiles
   where
-    loadFile :: FilePath -> IO (Either LoadRulesError DomainDefinition)
+    loadFile :: FilePath -> IO (Either LoadRulesError (Maybe DomainDefinition))
     loadFile file = catch
-      ((mapLeft (LoadRulesParseError file)) <$> Yaml.decodeFileEither @DomainDefinition file)
+      (parseErrorHandle file <$> Yaml.decodeFileEither @DomainDefinition file)
       (pure . Left . LoadRulesIOError)
 
+    -- | Handle a special case when the input file cannot be read due
+    -- to file permissions by returning Nothing on the Right.
+    parseErrorHandle
+      :: FilePath
+      -> Either Yaml.ParseException DomainDefinition
+         ----------------------------------------------
+      -> Either LoadRulesError (Maybe DomainDefinition)
+    parseErrorHandle _    (Right def)  = Right $ Just def
+    parseErrorHandle file (Left parEx) = case parEx of
+      Yaml.InvalidYaml (Just (Yaml.YamlException _)) ->
+        Right Nothing
+      err ->
+        Left $ LoadRulesParseError file err
```

Comment:
Hm, so you built #103 on top of master and not #102?

Reply:
No, it was the other way around. Nevertheless, PR #102 got merged in the meantime and casing as above is not there anymore.